user/stubs.c

```diff
@@ -0,0 +1,119 @@
+#include "userargs.h"
+#include <sys/syscall.h>
+
+sgxlkl_userargs_t* __sgxlkl_userargs;
+
+void sgxlkl_warn(const char* fmt, ...);
+
+int snprintf(char *str, size_t size, const char *format, ...);
+
+/*
+**==============================================================================
+**
+** syscall:
+**
+**==============================================================================
+*/
+
+long lkl_syscall(long no, long* params)
+{
+    long ret = __sgxlkl_userargs->ua_lkl_syscall(no, params);
+
+    return ret;
+}
+
+long __sgxlkl_log_syscall(
+    // sgxlkl_syscall_kind type,
+    uint32_t type,
+    long n,
+    long res,
+    int params_len,
+    ...)
+{
+    sgxlkl_warn("__sgxlkl_log_syscall() unimplemented in user space");
+    return 0;
+}
+
+/*
+**==============================================================================
+**
+** bypasses:
+**
+**==============================================================================
+*/
+
+void sgxlkl_warn(const char* msg, ...)
+{
+    /* ATTN: ignore variadic arguments */
+    return __sgxlkl_userargs->ua_sgxlkl_warn(msg);
+}
+
+void sgxlkl_error(const char* msg, ...)
+{
+    /* ATTN: ignore variadic arguments */
+    return __sgxlkl_userargs->ua_sgxlkl_error(msg);
+}
+
+void sgxlkl_fail(const char* msg, ...)
+{
+    /* ATTN: ignore variadic arguments */
+    return __sgxlkl_userargs->ua_sgxlkl_fail(msg);
+}
```

Comment:
We shouldn't be calling any kernel code other than the syscall interface in userspace.  Please can you conditionally compile these so that they appear only in debug builds?